# Calculate Measures of Central Tendency - Mean, Median, and Mode for the points & price columns and print their values using DataFrame Functions 

In [1]:
#importing necessary libraries and data
import numpy as np
import pandas as pd
data = pd.read_csv('../input/winereviewscsv/winemag-data_first150k.csv')

In [2]:
#calculating mean
data[['points','price']].mean()

points    87.888418
price     33.131482
dtype: float64

From above, you can see that the Mean or Average points ( or score) in the wine dataset is 87 which tells us  that Most Wines in the Dataset are Highly-Rated & average price of a wine bottle in the dataset is 33

In [3]:
#calculating median
data[['points','price']].median()

points    88.0
price     24.0
dtype: float64

From above, you can see that the Median price of a Wine bottle in the dataset is 24 which tells us that Atleast half of the 
wines in the dataset are sold for 24 or less & Median points in the dataset is 88 which tells us that Atleast half of the wines 
in the dataset received 88 or less points

In [4]:
#calculating mode
md = data[['points','price']].mode() 
print(md) 

   points  price
0      87   20.0


From above, you can see that the Mode is close to the median. Mode is defined as the value that appears 
the most frequently in our data.

# Calculate Measures of Spread - Range, Standard Deviation, and Variance for the points & price columns  and print their values

In [5]:
min_price = min(data['price']) 
max_price = max(data['price']) 
price_range = max_price - min_price 
print(price_range) 

2296.0


From above, you can see that the range is 2296 which tells us that the data is extremely is spread out 
considering the Median price of a wine bottle in the dataset is 24 
Similarly, try calculating the range for the points column

In [6]:
#calculating standard deviation
np.std(data[['points', 'price']], axis=0)

points     3.222381
price     36.322404
dtype: float64

Standard Deviation tells us how much the data varies from the mean 

• From above, you can see that the Standard Deviation for points is 3.222 which is obvious because
points only range from 80 to 100

• You can also see that the Standard Deviation for price is much higher because of its Outliers. The
higher the standard deviation, the more spread out the data is around the mean and vice-versa.

In [7]:
#calculating varaince
np.var(data[['points', 'price']], axis=0) 

points      10.383740
price     1319.317022
dtype: float64

From above, you can see that the Variance for points & price columns are 10.383740 and 1319.317022 
• Variance is a measure of how far each observation in the dataset is from the Mean
• It is the Square of Standard Deviation

# Obtain different summary statistics (Skewness and Kurtosis) for the points column and print their values

In [8]:
from scipy.stats import kurtosis
print("pearson kurtosis", kurtosis(data['points'], fisher=False))
print("pearson kurtosis", kurtosis(data['points'], fisher=True))

pearson kurtosis 2.713553658335501
pearson kurtosis -0.2864463416644991


It is the Measure of peaked-ness relative to a Gaussian shape 
scipy.stats.kurtosis(array, axis=0, fisher=True, bias=True) function calculates the kurtosis (Fisher or 
Pearson) of a data set. 
Returns Kurtosis value of the Normal Distribution for the Dataset 
High kurtosis in a data set is an indicator that data has heavy tails or outliers

In [9]:
from scipy.stats import skew 
skew(data['points'], axis=0)

0.14283121423675518

From above, you can see that the skewness is 0.14 which is> 0 which indicates that it is left skewed 
Skewness is the Measure of Lack of Symmetry 
scipy.stats.skew{array, axis=O, bias=True) function calculates the skewness of the data set 
If skewness = 0, Normally Distributed 
skewness< 0, More weight in the right tail of distribution 
skewness > 0, More weight in the left tail of distribution 

# Obtain the Z-Scores for the points column - Intermediate 

In [10]:
from scipy import stats

z = np.abs(stats.zscore(data['points']))
print('z-scores based on data in points column\n')
print(z)

z-scores based on data in points column

0         2.517263
1         2.517263
2         2.517263
3         2.517263
4         2.206934
            ...   
150925    0.965616
150926    0.965616
150927    0.965616
150928    0.655286
150929    0.655286
Name: points, Length: 150930, dtype: float64


The abs() function returns the absolute value of the specified number.

# Obtain the IQR value for the points column

In [11]:
from scipy.stats import iqr 
iqr(data['points'],axis=0, interpolation='higher') 

4

The Interquartile range (IQR) is the difference between the 75th and 25th percentile of the data. It is a 
measure of the dispersion similar to standard deviation or variance, but is much more robust against outliers